backend/data/processors/programs_processing.py

```diff
@@ -219,6 +220,7 @@ def addDisciplineData(components, item, programData):
 
 
 
+# !TODO: add a docstring here if you understand this
```

Comment:
I've deleted this function on my branch, but yeah whoever originally wrote it didn't put docstrings

Reply:
sgtm;
we'll just leave it here for now and then it can get removed when your branch gets merged in